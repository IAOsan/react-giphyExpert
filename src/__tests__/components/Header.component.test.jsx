import { vi } from 'vitest';
import '@testing-library/jest-dom';
import { render, createUser, waitFor, fireEvent } from '../test-utils';
import Header from '../../components/Header.component';

function renderHeader(props = {}) {
	return render(<Header {...props} />);
}

describe('<Header />', () => {
	const user = createUser();
	const onSearchStub = vi.fn();
	let wrapper = renderHeader({ onSearch: onSearchStub });

	beforeEach(() => {
		wrapper = renderHeader({ onSearch: onSearchStub });
	});

	afterEach(() => {
		vi.clearAllMocks();
	});

	it('should be render correctly', () => {
		expect(wrapper.container.firstElementChild).toMatchSnapshot();
		expect(wrapper.getByText(/gifexpert app/i)).toBeInTheDocument();
		expect(wrapper.getByRole('textbox')).toBeInTheDocument();
	});
	it('should be submit form calling onSeach prop', async () => {
		const query = 'goku';

		await user.type(wrapper.getByRole('textbox'), query);
		fireEvent.submit(document.querySelector('form'));

		await waitFor(() => {
			expect(onSearchStub).toBeCalledTimes(1);
			expect(onSearchStub).toBeCalledWith(expect.any(Function)); // expects any function
		});
	});
	it('should not submit form if input field is empty', async () => {
		const query = '    ';

		await user.type(wrapper.getByRole('textbox'), query);
		fireEvent.submit(document.querySelector('form'));

		await waitFor(() => {
			expect(onSearchStub).toBeCalledTimes(0);
		});
	});
});
