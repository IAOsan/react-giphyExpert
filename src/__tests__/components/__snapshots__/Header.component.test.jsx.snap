// Vitest Snapshot v1

exports[`<Header /> > should be render correctly 1`] = `
<header
  class="bg-primary py-4"
>
  <div
    class="container"
  >
    <h1
      class="h2 text-light"
    >
      GifExpert App
    </h1>
    <form>
      <div
        class="form-group py-4"
      >
        <input
          class="form-control"
          name="query"
          required=""
          type="text"
          value=""
        />
      </div>
    </form>
  </div>
</header>
`;
